Dockerfile

```diff
@@ -292,16 +301,16 @@ RUN mkdir -p /opt/root/bin &&\
 # be set at runtime (or in the final docker stage for docker runs).
 COPY --from=collectd /usr/local/bin/patchelf /usr/bin/
 
-# Gather Python dependencies
-COPY --from=python-plugins /usr/lib/python2.7/ /opt/root/lib/python2.7
-COPY --from=python-plugins /usr/local/lib/python2.7/ /opt/root/lib/python2.7
-COPY --from=python-plugins /usr/bin/python /opt/root/bin/python
-
 # Gather compiled collectd plugin libraries
 COPY --from=collectd /usr/sbin/collectd /opt/root/bin/collectd
 COPY --from=collectd /opt/deps/ /opt/root/lib/
 COPY --from=collectd /usr/lib/collectd/*.so /opt/root/lib/collectd/
 
+# Gather Python dependencies
```

Comment:
The reason you had to move this block below the collectd COPYs is because collectd is still compiling with the old Python 2.7 in Ubuntu's repos. You are also pulling in `libpython.so` from the `collectd` stage first and then overridding it with `libpython.so` from the `python-plugins` stage.  You need to download/compile Python in the first stage and use the same thing in both stages (either move the `python-plugins` stage to above `collectd` and then COPY `libpython.so` and the headers into the `collectd` stage, or do the download/compile in the `collectd` stage and copy it into the `python-plugins` stage).

Reply:
Right now you are linking collectd with one version of Python and then running it against a different version, which is risky.

Reply:
Will try to get the collectd stage to build with the new python. Need to be careful not to conflict with all the other python dependencies installed by apt.

Reply:
If you compile python in the `collectd` stage, you can set the `--prefix` to something like `/opt/python` and do a `make && make install` and it will put it there without conflicts with system Python.  Then just set some flags in the `./configure` for collectd to point to the alternate Python installation.